jdbc-driver/src/main/java/com/google/cloud/dataproc/jdbc/DataprocInfo.java

```diff
@@ -200,13 +207,22 @@ public String formatClusterFilterString() throws InvalidURLException {
     public Cluster findClusterInPool(String filter) throws SQLException {
         try {
             Cluster suitableCluster = null;
+            Map<Cluster, Long> clusterLoads = new HashMap<>();
             for (Cluster response :
                     clusterControllerClient
                             .listClusters(params.projectId(), params.region(), filter)
                             .iterateAll()) {
-                // TODO: decide on how to pick a cluster among cluster pool
+                if (response.hasMetrics()
+                        && response.getMetrics().getYarnMetricsOrDefault(YARN_MEMORY, 0) != 0) {
+                    clusterLoads.put(response, response.getMetrics().getYarnMetricsOrDefault(YARN_MEMORY, 0));
+                }
                 suitableCluster = response;
```

Comment:
Should this line be removed?

Reply:
I keep this line so that if all clusters in the pool have 0 yarn memory available (i.e. clusterLoads is empty), one cluster will still be picked. Is that the case or do we need to throw SQL exception if no cluster has available memory?